### ISSUE TITLE
Log review interface.

### ISSUE THREAD

**Issue by aethelwulffe** [LibreHealthIO/LibreEHR#479]
The Administration->Logs feature, which lets you do searches in the logs by user etc...  should not do anything except display the interface when you navigate to it.  Currently, it executes queries, sometimes crashing marginal systems with a lot of data in the logs without the user ever having selected any search criteria.  Often times, this is due simply to accidentally selecting the wrong item, getting logs and having to wait until the system starts responding again.

**Comment by navendu29** [comment]
Can I work on this Usability feature ?

**Comment by aethelwulffe** [comment]
Yes you may, of course.  To assign you, I will need to add you to the collaborator list.

**Comment by aethelwulffe** [comment]
Are you a GSOC hopeful, or do you just want to be added to the standard team invitation?

**Comment by navendu29** [comment]
Yes, I intend to apply for GSOC .

**Comment by aethelwulffe** [comment]
Then you may accept the invitation I sent you to that team, and welcome aboard!

**Comment by navendu29** [comment]
Thank you for the invitation.

**Comment by navendu29** [comment]
What type of interface do u want? 
Should the interface only contain the log information from the log tabel ?

**Comment by aethelwulffe** [comment]
Ha!  I don't want an interface.  The community does.  That means you get to decide and design it, in light of it's common usage case....
  So, what is this log table used for?  

- It is used to track activity by users, sort of a "blame" track that can be used to identify a series of events.
- A source for extracting every query that has been made and use that do things like recreate the data after someone has deleted it.  On this piece, it would be nice if the log.comment section kept the queries clean and accessible instead of prepending them with other strings.
- It can be used to find where some types of upgrade or imports have happened for troubleshooting purposes.
Now, in regard to the above, and the fact that at the moment the log interface has a very small footprint, it would be nice if it got improved without turning it into a huge project unto itself, with a whole bunch of assets that only it uses.  While the criteria selection and display are a bit clunky, the main issue is about the fact that the page takes forever to load because it tries to do things right when you click on the menu item...sometimes taking up massive resources and freezing the server...which is the main objective of the issue.

**Comment by navendu29** [comment]
I am getting a trouble in implementing this feature. I think i need to study more regarding this for which i would need some time so , if anyone else wants to work on this feature he/she may proceed .

**Comment by tmccormi** [comment]
A pull request is coming from @growlingflea that is the baseline.  It's an Activity Log viewer we created for Dr Bob Hoyt.  Uses DataTables.

Looks like this
1) Default  TODAY
![defaultactivitytoday](https://user-images.githubusercontent.com/285835/41379733-a242fe6a-6f17-11e8-81ed-99aaa54e8f46.png)

2) Filtered Search
![filteredactivitylog](https://user-images.githubusercontent.com/285835/41379732-a225cce6-6f17-11e8-8c92-286505c98882.png)

3) Summary
![summaryactivitylog](https://user-images.githubusercontent.com/285835/41379734-a25b2648-6f17-11e8-99ac-1ab731ccb1a7.png)

4) Details
![activitylog](https://user-images.githubusercontent.com/285835/41379735-a2763168-6f17-11e8-8a8c-8534fcac291b.png)
